include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -7323,15 +7321,26 @@ FEEvaluation<dim,
   evaluate(const VectorizedArrayType *            values_array,
            const EvaluationFlags::EvaluationFlags evaluation_flags)
 {
-  SelectEvaluator<dim, fe_degree, n_q_points_1d, VectorizedArrayType>::evaluate(
-    n_components,
-    evaluation_flags,
-    *this->data,
-    const_cast<VectorizedArrayType *>(values_array),
-    this->values_quad,
-    this->gradients_quad,
-    this->hessians_quad,
-    this->scratch_data);
+  if (fe_degree > -1)
+    SelectEvaluator<dim, fe_degree, n_q_points_1d, VectorizedArrayType>::
+      evaluate(n_components,
+               evaluation_flags,
+               *this->data,
+               const_cast<VectorizedArrayType *>(values_array),
+               this->values_quad,
+               this->gradients_quad,
+               this->hessians_quad,
+               this->scratch_data);
+  else
+    internal::FEEvaluationFactory<dim, Number, VectorizedArrayType>::evaluate(
+      n_components,
+      evaluation_flags,
+      *this->data,
+      const_cast<VectorizedArrayType *>(values_array),
+      this->values_quad,
+      this->gradients_quad,
+      this->hessians_quad,
+      this->scratch_data);
```

Comment:
I agree, but let us solve this with #10928 because that would reduce the duplication to 3 (?) arguments, at which point duplication is very likely the more readable version compared to adding a layer to collect the two directions (e.g. lambda).

Reply:
I agree!